src/SignalR/clients/ts/FunctionalTests/yarn.lock

```diff
@@ -2,17 +2,13 @@
 # yarn lockfile v1
 
 
-"@aspnet/signalr-protocol-msgpack@file:../signalr-protocol-msgpack":
```

Comment:
Why change this file instead of deleting it (as you did src/Middleware/NodeServices/src/yarn.lock)?

Conversely, @BrennanConroy told me a while ago that we can't really delete these `.lock` files or avoid them changing when `./restore.cmd` is run locally, especially if users have `yarn` installed globally. If we can remove the files, wonderful but let's do it everywhere and add `yarn.lock` to our top-level `.gitignore`

Reply:
If you read carefully, I _added_ src/Middleware/NodeServices/src/yarn.lock and deleted src/Middleware/NodeServices/src/package-lock.json. In NodeJS, committing the lock file is a good thing because it makes builds more reliable. The dependency resolution algorithm for yarn and npm much less stable than nuget -- same inputs might result in different dependencies. 

This particular change is a reflection of the change I made in package.json to switch from 'file:' back to 'link:'. In yarn, 'link' is the way to do a "project reference".